test/core/formattersTests.ts

```diff
@@ -1,193 +1,280 @@
 ///<reference path="../testReference.ts" />
 
 describe("Formatters", () => {
-  describe("fixed", () => {
-    it("shows exactly [precision] digits", () => {
+  describe("fixed()", () => {
+    it("shows correct amount of digits according to precision", () => {
       let fixed3 = Plottable.Formatters.fixed();
-      let result = fixed3(1);
-      assert.strictEqual(result, "1.000", "defaults to three decimal places");
-      result = fixed3(1.234);
-      assert.strictEqual(result, "1.234", "shows three decimal places");
-      result = fixed3(1.2346);
-      assert.strictEqual(result, "1.235", "changed values are not shown (get turned into empty strings)");
-    });
+      assert.strictEqual(fixed3(1), "1.000", "shows three decimal places");
+      assert.strictEqual(fixed3(-1), "-1.000", "shows three decimal places");
+      assert.strictEqual(fixed3(1.234), "1.234", "shows three decimal places");
+      assert.strictEqual(fixed3(-1.234), "-1.234", "shows three decimal places");
+      assert.strictEqual(fixed3(1.2346111111), "1.235", "shows three decimal places");
+      assert.strictEqual(fixed3(-1.2346111111), "-1.235", "shows three decimal places");
+
+      assert.strictEqual(fixed3(Infinity), "Infinity", "formats non-numeric number correctly");
+      assert.strictEqual(fixed3(-Infinity), "-Infinity", "formats non-numeric number correctly");
+      assert.strictEqual(fixed3(NaN), "NaN", "formats non-numeric number correctly");
 
-    it("precision can be changed", () => {
       let fixed2 = Plottable.Formatters.fixed(2);
-      let result = fixed2(1);
-      assert.strictEqual(result, "1.00", "formatter was changed to show only two decimal places");
+      assert.strictEqual(fixed2(1), "1.00", "shows two decimal places");
+      assert.strictEqual(fixed2(-1), "-1.00", "shows two decimal places");
+      assert.strictEqual(fixed2(1.23), "1.23", "shows two decimal places");
+      assert.strictEqual(fixed2(-1.23), "-1.23", "shows two decimal places");
+      assert.strictEqual(fixed2(1.2346111111), "1.23", "shows two decimal places");
+      assert.strictEqual(fixed2(-1.2346111111), "-1.23", "shows two decimal places");
+
+      let fixed0 = Plottable.Formatters.fixed(0);
+      assert.strictEqual(fixed0(1), "1", "shows no decimal places");
+      assert.strictEqual(fixed0(-1), "-1", "shows no decimal places");
+      assert.strictEqual(fixed0(1.23), "1", "shows no decimal places");
+      assert.strictEqual(fixed0(-1.23), "-1", "shows no decimal places");
```

Comment:
The description of this test and line 28 is identical. Same for 27 & 30. If the test fails, there is no way to tell which of the asserts failed.